### ISSUE TITLE
Handling of missings

### ISSUE THREAD

**Issue by tlienart** [alan-turing-institute/MLJBase.jl#97]
Not sure if it's the appropriate place to post this; it looks like we could improve the situations when dealing with missings. Here's a simple example of something that's not ideal:

1. `y` has some missings
1. `X` doesn't
1. we drop the rows of `y`  and `X`  that have missings
1. we feed that in a machine
1. the machine fails because the model doesn't handle `Union{Missing,Float64}`

of course  there are no missings in `y`. Interestingly, the scientific type does not contain missing so that could be exploited

Maybe we  could help with:

* provide a `dropmissing` function which "re-coerces"
* have a mechanism in `fit` which checks whether the type and scientific type match in terms of missing, if they don't and ST has it right, maybe the data can be re-coerced

thoughts?

**Comment by ablaom** [comment]
Just a short response for now. The `dropmissing` functionality already exists in julia, no?

```julia
julia> y = [1.0, 2.0, missing][1:2]
2-element Array{Union{Missing, Float64},1}:
 1.0
 2.0

julia> skipmissing(y) |> collect
2-element Array{Float64,1}:
 1.0
 2.0
```

**Comment by tlienart** [comment]
I think this needs a bit more brainstorming, yes `skipmissing |> collect` will work but it's not what  I think we should be doing (it requires work from the user and it's arguably confusing that as a user if you drop rows, the type remains with missing, I got caught so people will get caught). 

I would recommend the following: (let `A` denote the situation described above where you have removed rows with missing but the type is still  `Union{Missing,T}`)

* before a `fit` is called, check whether `A`
* if `A` then effectively pass a squeezed copy of `y`

this will guarantee that the fit doesn't fail weirdly

this is what I meant with `dropmissing` but could be called `squeezemissing` or whatever.

**Note**: it's important to note  that users will likely not use the skipmissing collect mechanism because it loses track of row indices. This is why I think a bit more thought needs to be put into this. And  having to  use `skipmissing > collect` on a vector where we have already removed missing is annoying and counter intuitive and will definitely make users stumble.

**Comment by ablaom** [comment]
I agree this requires a little thought. Let's flag this issue for discussion at next meeting. Of course input welcome in the meantime.
